test/framework/src/main/java/org/elasticsearch/test/XContentTestUtils.java

```diff
@@ -197,13 +197,14 @@ public static BytesReference insertRandomFields(XContentType contentType, BytesR
 
         try (XContentParser parser = createParser(NamedXContentRegistry.EMPTY, xContent, contentType)) {
             Supplier<Object> value = () -> {
+                List<Object> randomValues = RandomObjects.randomStoredFieldValues(random, contentType).v1();
                 if (random.nextBoolean()) {
-                    return RandomObjects.randomStoredFieldValues(random, contentType);
+                    return randomValues.get(0);
```

Comment:
There was a bug, `RandomObjects.randomStoredFieldValues` returns a tuple of values (random/expected after parsing), we want only a single random value here. This wasn't evident in the tests so far because the values were then written to xContent as Strings ( the object reference "Tuple@dfhjks345345d"), but this corrects it.